cloud/aws/cli.py

```diff
@@ -157,104 +157,59 @@ def deploy_lambda_image(c):
     )
 
 
-@task
-def run_vast_task(c, cmd_override=None):
-    """Start a new VAST server task on Fargate. DANGER: might lead to inconsistant state"""
-    cluster = terraform_output(c, "step-2", "fargate_cluster_name")
-    subnet = terraform_output(c, "step-2", "ids_appliances_subnet_id")
-    sg = terraform_output(c, "step-2", "vast_security_group")
-    task_def = terraform_output(c, "step-2", "vast_task_definition")
-    overrides = {}
-    if cmd_override is not None:
-        overrides["containerOverrides"] = [
-            {
-                "name": "main",
-                "command": [cmd_override],
-            },
-        ]
-    task_res = aws("ecs").run_task(
-        cluster=cluster,
-        count=1,
-        enableECSManagedTags=True,
-        enableExecuteCommand=True,
-        propagateTags="TASK_DEFINITION",
-        launchType="FARGATE",
-        networkConfiguration={
-            "awsvpcConfiguration": {
-                "subnets": [subnet],
-                "securityGroups": [sg],
-                "assignPublicIp": "DISABLED",
-            }
-        },
-        taskDefinition=task_def,
-        overrides=overrides,
-    )
-    if len(task_res["failures"]) > 0:
-        print(
-            f'{AWS_REGION} might need to be "activated" by running an EC2 instance in it'
-        )
-        raise Exit(task_res["failures"], 1)
-    task_arn = task_res["tasks"][0]["taskArn"].split("/")[-1]
-    print(f"Started task {task_arn}")
-
-
 @task(autoprint=True)
-def get_vast_server(c):
+def get_vast_server(c, blocking=False):
     """Get the task id of the VAST server"""
     cluster = terraform_output(c, "step-2", "fargate_cluster_name")
     family = terraform_output(c, "step-2", "vast_task_family")
     task_res = aws("ecs").list_tasks(family=family, cluster=cluster)
     nb_vast_tasks = len(task_res["taskArns"])
-    if nb_vast_tasks == 0:
+    if nb_vast_tasks == 0 and blocking:
+        time.sleep(1)
+        return get_vast_server(c, blocking)
```

Comment:
come on, I never get to write recursions ! 😄 

this can loop infinitely and thus, in theory, overflow (except if python does some magic to avoid this, but I don't know about anything like this). But with 1 call by sec, it would probably take a few days (or months) to fill the memory 😃 

`max_wait_time` is a great idea! let me add it.

I haven't tried writing this with a loop, if you really don't like the recursive style I can give it a try 🥷

Reply:
actually with `max_wait_time` the iterative style works better, so I'll switch for that 😉